python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1,693 +1,626 @@
-        return self._apply_aggregation("min")
-
-    def max(self):
-        return self._apply_aggregation("max")
+import pandas as pd
 
-    def mean(self):
-        return self._apply_aggregation("mean")
-
-    def count(self):
-        return self._apply_aggregation("count")
-
-    def agg(self, func):
-        return self._apply_aggregation(func)
-
-    def size(self):
-        from cudf.core.column import column_empty
-
-        nrows = len(self._groupby.obj)
-        data = cudf.Series(column_empty(nrows, "int8", masked=False))
-        return data.groupby(self._groupby.key_columns).count()
+import cudf
+import cudf._libxx.groupby as libgroupby
 
-    def serialize(self):
-        header, frames = self._groupby.serialize()
-        header["type-serialized"] = pickle.dumps(type(self))
-        return header, frames
 
-    @classmethod
-    def deserialize(cls, header, frames):
-        groupby_type = pickle.loads(header["type-serialized"])
-        _groupby = _GroupbyHelper.deserialize(header, frames)
-        by = None
-        if _groupby.level is None:
-            by = _groupby.by
-        return groupby_type(
-            _groupby.obj,
-            by=by,
-            level=_groupby.level,
-            sort=_groupby.sort,
-            as_index=_groupby.as_index,
-        )
+class GroupBy(object):
+    """
+    Group a DataFrame or Series by a set of columns.
 
+    Parameters
+    ----------
+    by : optional
+        Specifies the grouping columns. Can be any of the following:
+        - A Python function called on each value of the object's index
+        - A dict or Series that maps index labels to group names
+        - A cudf.Index object
+        - A str indicating a column name
+        - An array of the same length as the object
+        - A Grouper object
+        - A list of the above
+    level : int, level_name or list, optional
+        For objects with a MultiIndex, `level` can be used to specify
+        grouping by one or more levels of the MultiIndex.
+    sort : True, optional
+        If True (default), sort results by group9s). Note that
+        unlike Pandas, this also sorts values within each group.
+    as_index : bool, optional
+        If as_index=True (default), the group names appear
+        as the keys of the resulting DataFrame.
+        If as_index=False, the groups are returned as ordinary
+        columns of the resulting DataFrame, *if they are named columns*.
+    sort : True, optional
+        If True (default), sort results by group9s). Note that
+        unlike Pandas, this also sorts values within each group.
+    dropna : bool, optional
+        If True (default), do not include the "null" group.
+    """
 
-class SeriesGroupBy(_Groupby):
     def __init__(
-        self,
-        sr,
-        by=None,
-        level=None,
-        method="hash",
-        sort=True,
-        as_index=None,
-        dropna=True,
+        self, obj, by=None, level=None, sort=True, as_index=True, dropna=True
```

Comment:
We either need a `method=None` in here or a `**kwargs` to not break existing groupby code. And then we should throw a `DeprecationWarning` that `method` is deprecated and will be removed in a future release.